stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -257,6 +257,8 @@ Vector `kv.second` will be placed on the `kv.first` diagonal.
 By default the matrix is square and its size is inferred
 from `kv`, but a non-square size `m`×`n` (padded with zeros as needed)
 can be specified by passing `m,n` as the first arguments.
+For repeated diagonal indices `kv.first` the values in the corresponding
```

Comment:
Just out of curiousity, why this behavior?

Reply:
I guess to be consistent with how the `sparse(I, J, V)` works by default. Otherwise the order in which you supply the diagonal values starts to matter.